client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -711,6 +712,16 @@ static Request createRepository(PutRepositoryRequest putRepositoryRequest) throw
         return request;
     }
 
+    static Request deleteRepository(DeleteRepositoryRequest deleteRepositoryRequest) {
+        String endpoint = new EndpointBuilder().addPathPartAsIs("_snapshot").addPathPartAsIs(deleteRepositoryRequest.name()).build();
```

Comment:
THe second part should use addPathPart and go through escaping, given that it's user provided.